addons/web/models/ir_http.py

```diff
@@ -147,6 +150,17 @@ def session_info(self):
                             'parent_id': comp.parent_id.id,
                         } for comp in user.company_ids
                     },
+                    'unallowed_ancestor_companies': {
+                        comp.id: {
+                            'id': comp.id,
+                            'name': comp.name,
+                            'sequence': comp.sequence,
+                            # We need sudo since a user may not have access to ancestor companies and/or other
+                            # child companies of their inaccessible ancestors.
+                            'child_ids': (comp.sudo().child_ids & all_companies_in_hierarchy_sudo).ids,
+                            'parent_id': comp.sudo().parent_id.id,
```

Comment:
`unallowed_ancestor_companies_sudo` is already sudo'd, so is `comp`

Reply:
Totally right ;)